File: blue/src/colors_compute/endpoint.py
"""Packaged endpoint agent artifact; does not perform endpoint operations."""
import json
from importlib.resources import files

def endpoint_agent(provider):
    recipes = json.loads(files('colors_compute').joinpath('provider-recipes.json').read_text())
    if not isinstance(provider, str) or not recipes.get(provider, {}).get('application_reserved_ip'):
        raise ValueError('unsupported compute endpoint capability')
    return {'filename': 'colors-compute-endpoint',
            'content': files('colors_compute').joinpath('endpoint-agent.py').read_text(),
            'credentials': recipes[provider]['endpoint_credentials']}

File: red/src/endpoint.ts
import {readFileSync} from 'node:fs';
import recipes from '../resources/provider-recipes.json';
export function endpoint_agent(provider:string) {
 if(typeof provider!=='string'||!Object.hasOwn(recipes,provider)||!(recipes as any)[provider].application_reserved_ip)throw Error('unsupported compute endpoint capability');
 return {filename:'colors-compute-endpoint',content:readFileSync(new URL('../resources/endpoint-agent.py',import.meta.url),'utf8'),credentials:structuredClone((recipes as any)[provider].endpoint_credentials)};
}
